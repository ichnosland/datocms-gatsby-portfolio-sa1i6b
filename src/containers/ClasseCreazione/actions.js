/*
 *
 * ClasseCreazione actions
 *
 */

import {
  CLASSE_CREAZIONE_SCUOLE_ATTIVE_FETCH,
  CLASSE_CREAZIONE_SCUOLE_ATTIVE_SET,
  CLASSE_CREAZIONE_GEO_PROVINCIA_FETCH,
  CLASSE_CREAZIONE_GEO_SET,
  CLASSE_CREAZIONE_GEO_COMUNI_FETCH,
  CLASSE_CREAZIONE_GEO_COMUNI_SET,
  CLASSE_CREAZIONE_DATA_SET,
  CLASSE_CREAZIONE_DATA_POST,
  CLASSE_CREAZIONE_RESET,
  CLASSE_CREAZIONE_SPINNER_SET,
  CLASSE_CREAZIONE_FEEDBACK_SET,
  CLASSE_CREAZIONE_FEEDBACK_RESET,
  CLASSE_CREAZIONE_GEO_SCUOLE_SET,
} from './constants';


export const classeCreazioneScuoleAttiveFetch = () => ({
  type: CLASSE_CREAZIONE_SCUOLE_ATTIVE_FETCH,
});

export const classeCreazioneScuoleAttiveSet = (payload) => ({
  type: CLASSE_CREAZIONE_SCUOLE_ATTIVE_SET,
  payload,
});

export const classeCreazioneGeoProvinciaFetch = () => ({
  type: CLASSE_CREAZIONE_GEO_PROVINCIA_FETCH,
});

export const classeCreazioneGeoSet = (payload) => ({
  type: CLASSE_CREAZIONE_GEO_SET,
  payload,
});

export const classeCreazioneGeoComuneFetch = (sigla) => ({
  type: CLASSE_CREAZIONE_GEO_COMUNI_FETCH,
  sigla,
});

export const classeCreazioneGeoComuneSet = (payload) => ({
  type: CLASSE_CREAZIONE_GEO_COMUNI_SET,
  payload,
});

export const classeCreazioneGeoScuoleSet = (payload) => ({
  type: CLASSE_CREAZIONE_GEO_SCUOLE_SET,
  payload,
});

export const classeCreazioneDataSet = (payload) => ({
  type: CLASSE_CREAZIONE_DATA_SET,
  payload,
});

export const classeCreazioneDataPost = (payload, history) => ({
  type: CLASSE_CREAZIONE_DATA_POST,
  payload,
  history,
});

export const classeCreazioneReset = () => ({
  type: CLASSE_CREAZIONE_RESET,
});

export const classeCreazioneSpinnerSet = (enable) => ({
  type: CLASSE_CREAZIONE_SPINNER_SET,
  enable,
});

export const classeCreazioneFeedbackSet = (enable, tipologia, messaggio) => ({
  type: CLASSE_CREAZIONE_FEEDBACK_SET,
  enable,
  tipologia,
  messaggio,
});

export const classeCreazioneFeedbackReset = () => ({
  type: CLASSE_CREAZIONE_FEEDBACK_RESET,
});
