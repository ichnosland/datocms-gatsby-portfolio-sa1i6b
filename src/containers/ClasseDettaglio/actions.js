/*
 *
 * ClasseDettaglio actions
 *
 */

import {
  CLASSE_DETTAGLIO_ESPELLI,
  CLASSE_DETTAGLIO_RESET,
  CLASSE_DETTAGLIO_DATA_FETCH,
  CLASSE_DETTAGLIO_CORSO_FETCH,
  CLASSE_DETTAGLIO_CONTENUTO_SET,
  CLASSE_DETTAGLIO_DISPLAY_SET,
  CLASSE_DETTAGLIO_DISPLAY_SORT,
  CLASSE_DETTAGLIO_FEEDBACK_SET,
  CLASSE_DETTAGLIO_FEEDBACK_RESET,
  CLASSE_DETTAGLIO_SPINNER_SET,
} from './constants';

export const classeDettaglioEspelli = (payload) => ({
  type: CLASSE_DETTAGLIO_ESPELLI,
  payload,
});

export const classeDettaglioReset = () => ({
  type: CLASSE_DETTAGLIO_RESET,
});

export const classeDettaglioCorsoFetch = (payload) => ({
  type: CLASSE_DETTAGLIO_CORSO_FETCH,
  payload,
});

export const classeDettaglioDataFetch = (payload) => ({
  type: CLASSE_DETTAGLIO_DATA_FETCH,
  payload,
});

export const classeDettaglioContenutoSet = (payload) => ({
  type: CLASSE_DETTAGLIO_CONTENUTO_SET,
  payload,
});

export const classeDettaglioDisplaySort = (payloadData, sortingData) => ({
  type: CLASSE_DETTAGLIO_DISPLAY_SORT,
  payloadData,
  sortingData,
});

export const classeDettaglioDisplaySet = (payload) => ({
  type: CLASSE_DETTAGLIO_DISPLAY_SET,
  payload,
});

export const classeDettaglioFeedbackSet = (hasFeedback, tipologia, messaggio) => ({
  type: CLASSE_DETTAGLIO_FEEDBACK_SET,
  hasFeedback,
  tipologia,
  messaggio,
});

export const classeDettaglioFeedbackReset = () => ({
  type: CLASSE_DETTAGLIO_FEEDBACK_RESET,
});

export const classeDettaglioSpinnerSet = (enable) => ({
  type: CLASSE_DETTAGLIO_SPINNER_SET,
  enable,
});
