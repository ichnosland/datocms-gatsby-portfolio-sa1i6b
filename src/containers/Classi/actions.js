/*
 *
 * Classi actions
 *
 */

import {
  CLASSI_DATA_FETCH,
  CLASSI_DATA_SET,
  CLASSI_DATA_RESET,
  CLASSI_FEEDBACK_RESET,
  CLASSI_FEEDBACK_SET,
  CLASSI_SPINNER_SET,
} from './constants';

export const classiDataFetch = (idDisciplina) => ({
  type: CLASSI_DATA_FETCH,
  idDisciplina,
});

export const classiDataSet = (payload) => ({
  type: CLASSI_DATA_SET,
  payload,
});

export const classiDataReset = () => ({
  type: CLASSI_DATA_RESET,
});

export const classiFeedbackReset = () => ({
  type: CLASSI_FEEDBACK_RESET,
});

export const classiFeedbackSet = (enable = true, message = '', errorData = {}) => ({
  type: CLASSI_FEEDBACK_SET,
  enable,
  message,
  errorData,
});

export const classiSpinnerSet = (enable) => ({
  type: CLASSI_SPINNER_SET,
  enable,
});
