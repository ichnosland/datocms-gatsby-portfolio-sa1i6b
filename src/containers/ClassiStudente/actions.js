/*
 *
 * ClassiStudente actions
 *
 */

import {
  CLASSI_STUDENTE_FETCH,
  CLASSI_STUDENTE_SET,
  CLASSI_STUDENTE_RESET,
  CLASSI_STUDENTE_ISCRIVITI_POST,
  CLASSI_STUDENTE_SPINNER_SET,
  CLASSI_STUDENTE_FEEDBACK_SET,
  CLASSI_STUDENTE_FEEDBACK_RESET,
} from './constants';

export const classiStudenteFetch = (disciplinaId) => ({
  type: CLASSI_STUDENTE_FETCH,
  disciplinaId,
});

export const classiStudenteSet = (payload) => ({
  type: CLASSI_STUDENTE_SET,
  payload,
});

export const classiStudenteReset = () => ({
  type: CLASSI_STUDENTE_RESET,
});

export const classiStudenteIscrivitiPost = (codiceCorso, configuration) => ({
  type: CLASSI_STUDENTE_ISCRIVITI_POST,
  codiceCorso,
  configuration,
});

export const classiStudenteSpinnerSet = (enable) => ({
  type: CLASSI_STUDENTE_SPINNER_SET,
  enable,
});

export const classiStudenteFeedbackSet = (enable, tipologia, messaggio) => ({
  type: CLASSI_STUDENTE_FEEDBACK_SET,
  enable,
  tipologia,
  messaggio,
});

export const classiStudenteFeedbackReset = () => ({
  type: CLASSI_STUDENTE_FEEDBACK_RESET,
});
