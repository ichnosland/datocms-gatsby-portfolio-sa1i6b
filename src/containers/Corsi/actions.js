/*
 *
 * Corsi actions
 *
 */

import {
  CORSI_RESET,
  CORSI_ERROR_SET,
  CORSI_ERROR_RESET,
  CORSI_SPINNER_SET,
  CORSI_LIST_INITIALIZE,
  CORSI_LIST_DOCENTE_FETCH,
  CORSI_LIST_STUDENTE_FETCH,
  CORSI_LIST_SET,
  CORSI_CORSO_DETAIL_DOCENTE_TRIGGER,
  CORSI_CORSO_DETAIL_DOCENTE_SET,
} from './constants';

export const corsiReset = () => ({
  type: CORSI_RESET,
});

export const corsiSpinnerSet = (enable) => ({
  type: CORSI_SPINNER_SET,
  enable,
});

export const corsiErrorSet = (enable = true, message = '', errorData = {}) => ({
  type: CORSI_ERROR_SET,
  enable,
  message,
  errorData,
});

export const corsiErrorReset = () => ({
  type: CORSI_ERROR_RESET,
});

export const corsiListInitialize = (configuration, isDocente, userId) => ({
  type: CORSI_LIST_INITIALIZE,
  configuration,
  isDocente,
  userId,
});

export const corsiListDocenteFetch = (idDisciplina) => ({
  type: CORSI_LIST_DOCENTE_FETCH,
  idDisciplina,
});

export const corsiListStudenteFetch = (idDisciplina) => ({
  type: CORSI_LIST_STUDENTE_FETCH,
  idDisciplina,
});

export const corsiListSet = (payload) => ({
  type: CORSI_LIST_SET,
  payload,
});

export const corsiCorsoSelectedDocenteTrigger = (idCorso, configuration, isDocente, userId) => ({
  type: CORSI_CORSO_DETAIL_DOCENTE_TRIGGER,
  idCorso,
  configuration,
  isDocente,
  userId,
});

export const corsiCorsoDetailDocenteSet = (payload) => ({
  type: CORSI_CORSO_DETAIL_DOCENTE_SET,
  payload,
});
