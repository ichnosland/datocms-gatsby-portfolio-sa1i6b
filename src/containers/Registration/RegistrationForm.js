/**
*
* RegistrationForm
*
*/

import React from 'react';
import PropTypes from 'prop-types';
import { Field, reduxForm } from 'redux-form/immutable';
import FacebookLogin from 'react-facebook-login/dist/facebook-login-render-props';

import { FACEBOOK } from 'configuration';
import {
  CardForm,
  FormElement,
  FacebookButton,
  FacebookIcon,
} from 'components/FormElements';
import renderedField from 'components/FormElements/Field';
import { LegalInfo, Separator, SeparatorText } from 'components/Paragraph';
import { Button } from 'components/Button';
import LogoPlatform from 'components/LogoPlatform';
import icon from 'icons/buttons';
import validate from './validator';

export const required = (value) => value ? undefined : 'Obbligatorio';

export class Registration extends React.Component { // eslint-disable-line react/prefer-stateless-function
  render() {
    const { invalid, product, handleSubmit, responseFacebookFunction } = this.props;

    return (
      <CardForm onSubmit={handleSubmit}>
        <FormElement margin="0 auto 36px">
          <LogoPlatform
            product={product}
            formLogo
          />
        </FormElement>
        <FormElement>
          <FacebookLogin
            appId={FACEBOOK.appId}
            language={FACEBOOK.language}
            version={FACEBOOK.version}
            callback={responseFacebookFunction}
            isMobile
            disableMobileRedirect
            fields="name,email"
            scope="public_profile,email"
            render={/* istanbul ignore next */ (renderProps) => (
              <FacebookButton full onClick={renderProps.onClick} type="button">
                <FacebookIcon {...icon.facebook} left /> Registrati con Facebook
              </FacebookButton>
            )}
          />
        </FormElement>
        <Separator>
          <SeparatorText>oppure</SeparatorText>
        </Separator>
        <FormElement>
          <Field name="email" component={renderedField} type="email" label="Indirizzo email" validate={[required]} />
        </FormElement>
        <FormElement>
          <Field name="password" component={renderedField} type="password" label="Scegli una password" validate={[required]} />
        </FormElement>
        <FormElement>
          <Field name="password2" component={renderedField} type="password" label="Verifica la password" validate={[required]} />
        </FormElement>
        <FormElement>
          <LegalInfo>
            <Field
              name="privacy"
              id="privacy"
              component="input"
              type="checkbox"
              validate={[required]}
            />
            &nbsp;
            <label htmlFor="privacy">&nbsp; Ho letto ed accetto le <a href="https://maieuticallabs.it/privacy" target="_blank">normative sulla privacy</a></label>
          </LegalInfo>
        </FormElement>
        <FormElement>
          <Button actioncolor="okay" full type="submit" disabled={invalid}>Invia</Button>
        </FormElement>
      </CardForm>
    );
  }
}

Registration.propTypes = {
  handleSubmit: PropTypes.func.isRequired,
  invalid: PropTypes.bool,
  product: PropTypes.string,
  responseFacebookFunction: PropTypes.func.isRequired,
};

const RegistrationForm = reduxForm({
  form: 'registration',
  validate,
  asyncBlurFields: ['username'],
})(Registration);

export default RegistrationForm;
