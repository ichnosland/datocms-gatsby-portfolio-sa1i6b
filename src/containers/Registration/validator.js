export default (values) => {
  const errors = {};

  if (!values.get('email')) {
    errors.email = 'Campo obbligatorio';
  } else if (!/^[\w.-]+(\+\w+)?@[\w.-]+\.[\w.]{2,}$/i.test(values.get('email'))) {
    errors.email = 'Indirizzo email non valido';
  }

  if (!values.get('password')) {
    errors.password = 'Campo obbligatorio';
  } else if (values.get('password').length < 10) {
    errors.password = 'Deve essere di almeno 10 caratteri';
  }

  if (!values.get('password2')) {
    errors.password2 = 'Campo obbligatorio';
  } else if (values.get('password') !== values.get('password2')) {
    errors.password2 = 'Le password non coincidono';
  }

  return errors;
};
