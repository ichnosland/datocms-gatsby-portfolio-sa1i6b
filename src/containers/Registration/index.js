/*
 *
 * Registration
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import ZendeskTicket from 'containers/ZendeskTicket';
import ButtonGroup from 'components/ButtonGroup';
import Spinner from 'components/Spinner';
import Div from 'components/Div';
import AlertBanner from 'components/AlertBanner';
import ClearLink from 'components/ClearLink';
import P, { LegalInfo } from 'components/Paragraph';
import { FormSection, FormElement } from 'components/FormElements';
import { modalSetData } from 'containers/ModalBox/actions';
import { userRegistrationPost, userDataFetch } from 'containers/User/actions';
import RegistrationForm from './RegistrationForm';


export class RegistrationView extends React.PureComponent { // eslint-disable-line react/prefer-stateless-function
  onSubmit = (e) => {
    e.preventDefault();
    const { values: registrationData } = this.props.formRegistrationData;
    const { history, onUserRegister, configuration } = this.props;

    onUserRegister({
      ...registrationData,
      history,
      configuration,
    });
  }

  onResponseFacebook = (response) => {
    const { history, configuration, onUserLogin } = this.props;

    onUserLogin(
      {
        email: response.email,
        fbname: response.name,
        fbuid: response.id,
        fbtoken: response.accessToken,
        register: true,
      },
      configuration,
      history,
      true
    );
  }

  render() {
    const {
      configuration,
      onModalSetData,
      spinner,
      error,
    } = this.props;

    return (
      <FormSection paddingmobile="20px" className="Wrap">
        {[
          error.hasErrors && error.errorMessage && (
            <AlertBanner
              key="AlertBanner"
              text={error.errorMessage}
              actioncolor="error"
              margin="0 0 20px"
            >
              {error.errorMessage}
            </AlertBanner>
          ),
          spinner ?
            <Spinner key="Spinner" /> :
            [
              <Div
                align="center"
                key="Div_centrato"
              >
                <RegistrationForm
                  handleSubmit={this.onSubmit}
                  product={configuration.product}
                  responseFacebookFunction={this.onResponseFacebook}
                />
                <FormElement>
                  <P><ClearLink to="/login">Sei gi√† registrato?</ClearLink></P>
                </FormElement>
                <FormElement>
                  <ButtonGroup
                    buttons={[{
                      id: 1,
                      label: 'Assistenza',
                      onClickFunction: () => onModalSetData({
                        contenuto: (
                          <ZendeskTicket
                            ticketData={{
                              provenienza: 'registration',
                            }}
                          />
                        ),
                        show: true,
                        disableClose: true,
                        isPopup: false,
                      }),
                    }]}
                  />
                </FormElement>
              </Div>,
              <LegalInfo
                key="LegalInfo"
                align="center"
              >
                <a href="https://maieuticallabs.it/privacy/" target="_blank">Termini, condizioni d&apos;uso e normative sulla privacy</a><br className="mobileOnly" /> di <a href="https://maieuticallabs.it/" target="_blank">Maieutical Labs</a>.
              </LegalInfo>,
            ],
        ]}
      </FormSection>
    );
  }
}

RegistrationView.propTypes = {
  configuration: PropTypes.shape({
    product: PropTypes.string.isRequired,
  }).isRequired,
  spinner: PropTypes.bool.isRequired,
  error: PropTypes.object.isRequired,
  history: PropTypes.shape({
    push: PropTypes.func.isRequired,
  }).isRequired,
  onModalSetData: PropTypes.func.isRequired,
  onUserRegister: PropTypes.func.isRequired,
  formRegistrationData: PropTypes.object,
  onUserLogin: PropTypes.func.isRequired,
};

const mapDispatchToProps = (dispatch) => ({
  onModalSetData: (payload) => {
    dispatch(modalSetData(payload));
  },
  onUserRegister: (payload) => {
    dispatch(userRegistrationPost(payload));
  },
  onUserLogin: (loginData, configuration, history, isFacebook) => {
    dispatch(userDataFetch(loginData, configuration, history, isFacebook));
  },
});

const mapStateToProps = (state) => ({
  configuration: state.get('configuration').toJS(),
  userAuthentication: state.get('user').toJS().authentication,
  spinner: state.get('user').toJS().spinner,
  error: state.get('user').toJS().error,
  formRegistrationData: state.get('form').toJS().registration,
});

const Registration = connect(
  mapStateToProps,
  mapDispatchToProps
)(RegistrationView);

export default Registration;

