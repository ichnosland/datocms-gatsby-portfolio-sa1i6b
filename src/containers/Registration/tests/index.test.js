import React from 'react';
import { shallow } from 'enzyme';
import { connect } from 'react-redux';
import { compose } from 'redux';
import configureStore from 'configureStore';

import injectReducer from 'utils/injectReducer';
import reducerCurryer from 'common/applications/reducer';

import userReducer from 'containers/User/reducer';
import { modalSetData } from 'containers/ModalBox/actions';
import { userRegistrationPost, userDataFetch } from 'containers/User/actions';
import ButtonGroup from 'components/ButtonGroup';
import ZendeskTicket from 'containers/ZendeskTicket';
import Spinner from 'components/Spinner';
import AlertBanner from 'components/AlertBanner';
import Registration, { RegistrationView } from '../index';
import RegistrationForm from '../RegistrationForm';

const mockConfiguration = {
  product: 'product',
  hasPremium: true,
  disciplinaId: 123,
  homePage: '/home',
};
const history = {
  push: () => { },
};
const store = configureStore({}, {});
const mockPros = {
  spinner: false,
  error: {},
  history,
  configuration: mockConfiguration,
  onModalSetData: () => { },
  onUserRegister: () => { },
  onUserLogin: () => { },
  userAuthentication: {
    logged: false,
  },
  store,
};

describe('<RegistrationView />', () => {
  it('should render one Registration component', () => {
    const renderedComponent = shallow(
      <RegistrationView {...mockPros} />
    );

    expect(renderedComponent.find(RegistrationForm).length).toBe(1);
  });

  it('should render one Registration component', () => {
    const props = {
      ...mockPros,
      onModalSetData: jest.fn(),
    };
    const renderedComponent = shallow(
      <RegistrationView {...props} />
    );

    expect(props.onModalSetData).not.toHaveBeenCalled();
    const onClickButtonFx = renderedComponent.find(ButtonGroup).props().buttons[0].onClickFunction;

    onClickButtonFx();
    expect(props.onModalSetData).toHaveBeenCalledWith({
      contenuto: (
        <ZendeskTicket
          ticketData={{
            provenienza: 'registration',
          }}
        />
      ),
      show: true,
      disableClose: true,
      isPopup: false,
    });
  });

  it('onResponseFacebook should call onUserLogin', () => {
    const props = {
      ...mockPros,
      onUserLogin: jest.fn(),
    };
    const renderedComponent = shallow(
      <RegistrationView {...props} />
    );

    expect(props.onUserLogin).not.toHaveBeenCalled();
    renderedComponent.instance().onResponseFacebook({
      email: 'acme@example.com',
      name: 'nome',
      id: 12345,
      accessToken: 'accessToken',
    });
    expect(props.onUserLogin).toHaveBeenCalledWith(
      {
        email: 'acme@example.com',
        fbname: 'nome',
        fbuid: 12345,
        fbtoken: 'accessToken',
        register: true,
      },
      mockConfiguration,
      history,
      true
    );
  });

  it('should render one Registration component when spinner = true', () => {
    const props = {
      ...mockPros,
      spinner: true,
    };
    const renderedComponent = shallow(
      <RegistrationView {...props} />
    );

    expect(renderedComponent.find(Spinner).length).toBe(1);
    expect(renderedComponent.find(AlertBanner).length).toBe(0);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render one Registration component when hasErrors = true', () => {
    const props = {
      ...mockPros,
      error: {
        hasErrors: true,
        errorMessage: 'message',
      },
    };
    const renderedComponent = shallow(
      <RegistrationView {...props} />
    );

    expect(renderedComponent.find(Spinner).length).toBe(0);
    expect(renderedComponent.find(AlertBanner).length).toBe(1);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('onSubmit should call onUserRegister', () => {
    const props = {
      ...mockPros,
      onUserRegister: jest.fn(),
      formRegistrationData: {
        values: {
          username: 'username',
          password: 'password',
          password2: 'password2',
          email: 'email',
        },
      },
    };
    const renderedComponent = shallow(
      <RegistrationView {...props} />
    );
    const instance = renderedComponent.instance();

    instance.onSubmit({ preventDefault: () => { } });
    expect(props.onUserRegister).toHaveBeenCalledWith({
      configuration: mockConfiguration,
      history,
      ...props.formRegistrationData.values,
    });
  });
});

describe('<Registration />', () => {
  const withReducerConfiguration = injectReducer({ key: 'configuration', reducer: reducerCurryer(mockConfiguration) });
  const withReducerUser = injectReducer({ key: 'user', reducer: userReducer });

  const ConfigurationWrapper = compose(
    withReducerConfiguration,
    withReducerUser,
    connect()
  )(Registration);

  it('mapDispatchToProps > onModalSetData', () => {
    const mockStore = store;
    mockStore.dispatch = jest.fn();
    const renderedComponent = shallow(
      <ConfigurationWrapper
        match={{ params: { id: '123' } }}
        history={history}
      />, { context: { store: mockStore } }
    ).dive().dive().dive();

    expect(mockStore.dispatch).not.toHaveBeenCalledWith(modalSetData({ data: 123 }));
    renderedComponent.props().onModalSetData({ data: 123 });
    expect(mockStore.dispatch).toHaveBeenCalledWith(modalSetData({ data: 123 }));
  });

  it('onUserRegister > userRegistrationPost', () => {
    const mockStore = store;
    mockStore.dispatch = jest.fn();
    const renderedComponent = shallow(
      <ConfigurationWrapper
        match={{ params: { id: '123' } }}
        history={history}
      />, { context: { store: mockStore } }
    ).dive().dive().dive();

    expect(mockStore.dispatch).not.toHaveBeenCalledWith(userRegistrationPost({ data: 123 }));
    renderedComponent.props().onUserRegister({ data: 123 });
    expect(mockStore.dispatch).toHaveBeenCalledWith(userRegistrationPost({ data: 123 }));

    expect(mockStore.dispatch).not.toHaveBeenCalledWith(userDataFetch(1, 2, 3, 4));
    renderedComponent.props().onUserLogin(1, 2, 3, 4);
    expect(mockStore.dispatch).toHaveBeenCalledWith(userDataFetch(1, 2, 3, 4));
  });
});
