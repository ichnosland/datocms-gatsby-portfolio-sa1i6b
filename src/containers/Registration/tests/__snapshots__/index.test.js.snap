// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RegistrationView /> should render one Registration component when hasErrors = true 1`] = `
<FormElements__FormSection
  className="Wrap"
  paddingmobile="20px"
>
  <AlertBanner
    actioncolor="error"
    key="AlertBanner"
    margin="0 0 20px"
    text="message"
    theme={
      Object {
        "brand": "#00BBEF",
        "radius": Object {
          "general": "6px",
        },
      }
    }
  >
    message
  </AlertBanner>
  <Div
    align="center"
    key="Div_centrato"
  >
    <ReduxForm
      handleSubmit={[Function]}
      product="product"
      responseFacebookFunction={[Function]}
    />
    <FormElements__FormElement>
      <Paragraph
        theme={
          Object {
            "brand": "#00BBEF",
            "darken": "#1588AA",
          }
        }
      >
        <ClearLink
          to="/login"
        >
          Sei gi√† registrato?
        </ClearLink>
      </Paragraph>
    </FormElements__FormElement>
    <FormElements__FormElement>
      <ButtonGroup
        buttons={
          Array [
            Object {
              "id": 1,
              "label": "Assistenza",
              "onClickFunction": [Function],
            },
          ]
        }
      />
    </FormElements__FormElement>
  </Div>
  <Paragraph__LegalInfo
    align="center"
    key="LegalInfo"
  >
    <a
      href="https://maieuticallabs.it/privacy/"
      target="_blank"
    >
      Termini, condizioni d'uso e normative sulla privacy
    </a>
    <br
      className="mobileOnly"
    />
     di 
    <a
      href="https://maieuticallabs.it/"
      target="_blank"
    >
      Maieutical Labs
    </a>
    .
  </Paragraph__LegalInfo>
</FormElements__FormSection>
`;

exports[`<RegistrationView /> should render one Registration component when spinner = true 1`] = `
<FormElements__FormSection
  className="Wrap"
  paddingmobile="20px"
>
  <Spinner
    key="Spinner"
    number={3}
  />
</FormElements__FormSection>
`;
